From libc-ports-return-1326-listarch-libc-ports=sources dot redhat dot com at sourceware dot org Tue Aug 25 12:39:19 2009
Return-Path: <libc-ports-return-1326-listarch-libc-ports=sources dot redhat dot com at sourceware dot org>
Delivered-To: listarch-libc-ports at sources dot redhat dot com
Received: (qmail 25957 invoked by alias); 25 Aug 2009 12:39:19 -0000
Received: (qmail 25949 invoked by uid 22791); 25 Aug 2009 12:39:18 -0000
X-SWARE-Spam-Status: No, hits=-1.6 required=5.0 	tests=AWL,BAYES_00,J_CHICKENPOX_41,SPF_PASS
X-Spam-Check-By: sourceware.org
Received: from mail.codesourcery.com (HELO mail.codesourcery.com) (65.74.133.4)     by sourceware.org (qpsmtpd/0.43rc1) with ESMTP; Tue, 25 Aug 2009 12:39:09 +0000
Received: (qmail 16889 invoked from network); 25 Aug 2009 12:39:07 -0000
Received: from unknown (HELO mbp.local) (maxim@127.0.0.2)   by mail.codesourcery.com with ESMTPA; 25 Aug 2009 12:39:07 -0000
Message-ID: <4A93DB61.6070304@codesourcery.com>
Date: Tue, 25 Aug 2009 16:38:57 +0400
From: Maxim Kuvyrkov <maxim at codesourcery dot com>
User-Agent: Thunderbird 2.0.0.23 (Macintosh/20090812)
MIME-Version: 1.0
To: libc-ports at sourceware dot org
CC: Andreas Schwab <schwab at linux-m68k dot org>
Subject: [M68K/ColdFire patch 2/n] Add CFI information to dl-trampoline.S
References: <4A93D885.3060004@codesourcery.com>
In-Reply-To: <4A93D885 dot 3060004 at codesourcery dot com>
Content-Type: multipart/mixed;  boundary="------------050005000008050101020406"
X-IsSubscribed: yes
Mailing-List: contact libc-ports-help at sourceware dot org; run by ezmlm
Precedence: bulk
List-Id: <libc-ports.sourceware.org>
List-Subscribe: <mailto:libc-ports-subscribe at sourceware dot org>
List-Post: <mailto:libc-ports at sourceware dot org>
List-Help: <mailto:libc-ports-help at sourceware dot org>, <http://sourceware dot org/lists dot html#faqs>
Sender: libc-ports-owner at sourceware dot org
Delivered-To: mailing list libc-ports at sourceware dot org

This is a multi-part message in MIME format.
--------------050005000008050101020406
Content-Type: text/plain; charset=ISO-8859-1; format=flowed
Content-Transfer-Encoding: 7bit

This patch adds frame unwind information to _dl_runtime_resolve() to 
make debugging of the resolver more pleasant.

Any comments?  OK to commit?

Thanks,

--
Maxim Kuvyrkov
CodeSourcery

--------------050005000008050101020406
Content-Type: text/plain; x-mac-type="0"; x-mac-creator="0";
 name="0002-Add-frame-unwind-information-to-_dl_runtime_resolve.patch"
Content-Transfer-Encoding: 7bit
Content-Disposition: inline;
 filename*0="0002-Add-frame-unwind-information-to-_dl_runtime_resolve.pat";
 filename*1="ch"

>From 6db3a545be78373ca431e982fd354c8315e2519a Mon Sep 17 00:00:00 2001
From: Maxim Kuvyrkov <maxim@codesourcery.com>
Date: Mon, 24 Aug 2009 12:29:56 +0400
Subject: [PATCH 2/5] Add frame unwind information to _dl_runtime_resolve.

	* sysdeps/m68k/dl-trampoline.S: Add frame unwinding informarion.

This patch adds frame unwind information to _dl_runtime_resolve.
The initial CFA offset of 12 was determined by trial and error and
with debugger's assistance.
---
 sysdeps/m68k/dl-trampoline.S |   11 ++++++++++-
 1 files changed, 10 insertions(+), 1 deletions(-)

diff --git a/sysdeps/m68k/dl-trampoline.S b/sysdeps/m68k/dl-trampoline.S
index e324da1..1a46b1e 100644
--- a/ports/sysdeps/m68k/dl-trampoline.S
+++ b/ports/sysdeps/m68k/dl-trampoline.S
@@ -1,5 +1,5 @@
 /* PLT trampolines.  m68k version.
-   Copyright (C) 2005 Free Software Foundation, Inc.
+   Copyright (C) 2005, 2009 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -23,23 +23,32 @@
 	.globl _dl_runtime_resolve
 	.type _dl_runtime_resolve, @function
 _dl_runtime_resolve:
+	cfi_startproc
+	cfi_def_cfa (%sp, 12)
 	| Save %a0 (struct return address) and %a1.
 	move.l %a0, -(%sp)
+	cfi_adjust_cfa_offset (4)
 	move.l %a1, -(%sp)
+	cfi_adjust_cfa_offset (4)
 	| Call the real address resolver.
 	jbsr _dl_fixup
 	| Restore register %a0 and %a1.
 	move.l (%sp)+, %a1
+	cfi_adjust_cfa_offset (-4)
 	move.l (%sp)+, %a0
+	cfi_adjust_cfa_offset (-4)
 	| Pop parameters
 	addq.l #8, %sp
+	cfi_adjust_cfa_offset (-8)
 	| Call real function.
 #ifdef __mcoldfire__
 	move.l %d0,-(%sp)
+	cfi_adjust_cfa_offset (4)
 	rts
 #else
 	jmp (%d0)
 #endif
+	cfi_endproc
 	.size _dl_runtime_resolve, . - _dl_runtime_resolve
 
 	.text
-- 
1.6.4


--------------050005000008050101020406--

